pep-0572.rst

```diff
@@ -613,6 +620,8 @@ Broadly the same semantics as the current proposal, but spelled differently.
    confusion or require special-casing (eg to forbid assignment within the
    headers of these statements).
 
+   TODO: Explain more why this is bad, since this keeps coming up (it's also readability)
```

Comment:
I wish I could help with that, but I'm unable to articulate why `as` rubs me so much the wrong way.  Way back when, `as` was fine for import statements, perhaps because I didn't _think_ of them as "binding operations" to begin with - `as` just meant "rename this thing while importing it".  Later uses of `as` struck me as very minor warts, because they _were_ "about binding", but it was a fine compromise given that we didn't want the horror of introducing assignment expressions ;-)  Regardless, I dislike "as" for assignment expressions most of all the suggestions, because it violates "name on the left, value on the right" that's used "almost everywhere" in Python already.

I have noted that, in bind-and-test contexts,

    if name := f(x):

preserves that our eyes are already trained to "look to the far right" for the object being tested; the name is of lesser importance.  In code that didn't use assignment expressions for _everything_ (which is likely to be almost all code), it would be visually jarring to sometimes need to "look in the middle" instead for the important part:

    if f(x) as name:

But - ya - that sounds like a rationalization made up to justify a prejudice 😉

Reply:
My own intuition is that in `if foo(arg) as var` the assignment target doesn't jump out at you -- it just reads like `if foo arg blah blah` and it is too similar visually to `if foo(arg) and var`.

In all other situations where an `as TARGET` clause is allowed, even readers with intermediary skills are led to anticipate that clause (however optional) by the keyword that starts the line, and the grammar ties that keyword closely to the `as` clause:

- `import foo as bar`
- `except Exc as var`
- `with ctxmgr() as var`

To the contrary, the assignment expression does *not* belong to the `if` or `while` that starts the line, and we intentionally allow them in other contexts as well.

Also the parallel cadence between

- `VAR = EXPR`
- `if VAR := EXPR`

strikes me as reinforcing the visual recognition of assignment expressions.

Reply:
See #711